src/include/miopen/solver.hpp

```diff
@@ -370,12 +370,8 @@ struct PerformanceConfigConvAsm1x1U : PerfConfigBase<PerformanceConfigConvAsm1x1
 
     void StaticHeuristic(const miopen::conv::ProblemDescription& problem);
     void HeuristicInit(const ExecutionContext&, const miopen::conv::ProblemDescription&);
-#if MIOPEN_ENABLE_AI_KERNEL_TUNING
-    void RunParmeterPredictionModel(const ExecutionContext&,
-                                    const miopen::conv::ProblemDescription&,
-                                    bool& valid);
-    bool ModelApplyToken(int index, int value, const miopen::conv::ProblemDescription&);
-#endif
+    bool IsModelApplicable(const ExecutionContext& ctx,
+                           const miopen::conv::ProblemDescription& problem) const;
```

Comment:
```suggestion
#if MIOPEN_ENABLE_AI_KERNEL_TUNING
    bool IsModelApplicable(const ExecutionContext& ctx,
                           const miopen::conv::ProblemDescription& problem) const;
#endif
```

Reply:
I believe the above condition would be handled inside the method, such that if `MIOPEN_ENABLE_AI_KERNEL_TUNING` is not set, it will return false?

Reply:
I think this is implemented here without `#ifdef` to remove some `#ifdef`s in the code of the solver itself. But `RunParameterPredictionModel()` that is called after checking the condition is protected by `#ifdef` (as it should be), and there will be a compiler error if `MIOPEN_ENABLE_AI_KERNEL_TUNING` is not set.

Reply:
@averinevg I have gotten rid of the compiler error, the #ifdef is now placed in a more appropriate position.